xrpl/token.ts

```diff
@@ -0,0 +1,113 @@
+import { Command } from 'commander';
+
+import { printInfo } from '../common';
+import { hex } from './utils';
+
+const STANDARD_CURRENCY_CHARS = /^[A-Za-z0-9?!@#$%^&*<>(){}[\]|]+$/;
+
+function tokenSymbolToCurrencyCode(tokenSymbol: string): string {
+    if (tokenSymbol.length <= 3) {
+        if (!STANDARD_CURRENCY_CHARS.test(tokenSymbol)) {
+            throw new Error(
+                `Token symbol "${tokenSymbol}" contains invalid characters. Standard currency codes (≤3 chars) must only contain: letters, digits, and symbols: ?!@#$%^&*<>(){}[]|`,
+            );
+        }
+        return tokenSymbol;
+    }
+
+    const hexString = hex(tokenSymbol).toUpperCase();
+
+    if (hexString.length > 40) {
+        throw new Error(
+            `Token symbol "${tokenSymbol}" too long: hex representation (${hexString.length} chars) exceeds xrpl 40-character limit`,
+        );
+    }
+    const paddedHex = hexString + '0'.repeat(40 - hexString.length);
+
+    return paddedHex;
+}
+
+function currencyCodeToTokenSymbol(currencyCode: string): string {
+    if (currencyCode.length <= 3) {
+        if (!STANDARD_CURRENCY_CHARS.test(currencyCode)) {
+            throw new Error(
+                `Currency code "${currencyCode}" contains invalid characters. Standard currency codes (≤3 chars) must only contain: letters, digits, and symbols: ?!@#$%^&*<>(){}[]|`,
+            );
+        }
+        return currencyCode;
+    }
+
+    if (currencyCode.length !== 40) {
+        throw new Error(`Invalid currency code: ${currencyCode} must be exactly 40 characters (160-bit hex)`);
+    }
+
+    if (!/^[0-9A-Fa-f]+$/.test(currencyCode)) {
+        throw new Error(`Invalid currency code: ${currencyCode} is not a valid hex string`);
+    }
+
+    const trimmedHex = currencyCode.replace(/0+$/, '');
+
+    if (trimmedHex === '') {
+        throw new Error(`Invalid currency code: ${currencyCode} results in empty string after trimming trailing zeros`);
+    }
+
+    if (trimmedHex.length % 2 !== 0 || trimmedHex.length < 8) {
```

Comment:
```suggestion
    if (trimmedHex.length % 2 !== 0 || trimmedHex.length < 8) {
```
already covered by the length check here